### PR TITLE
Demographics fix

### PR DESCRIPTION
**Pull request by jawalonoski**
- Fixes a rare exception related to demographic distributions that are all zeroes (e.g., if every income bracket for a location has a zero probability -- then a person cannot exist within any of the income brackets, and a runtime exception is thrown).
- Rather than patch the demographics file, part of this fix is to detect all zero distributions and use a default uniform distribution (which was already being applied in the case where the distributions were null instead of zero). Don't like that? Override the demographics file.
- I also made partial fixes to the demographics file (fixed one county in New Mexico that was especially egregious in this regard).
- One other fix related to Unicode accented n's

### CODE REVIEW

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java [hunk 1]
```diff
@@ -502,12 +506,43 @@ private static Demographics csvLineToDemographics(Map<String,String> line) {
...
+   * @param map The map to check.
+   */
+  private static void nonZeroDefaults(Map<String, Double> map) {
+    // Any null or nan values should be zero
+    for (String key : map.keySet()) {
+      Double value = map.get(key);
+      if (value == null || value.isNaN()) {
+        map.put(key, Double.valueOf(0));
+      }
+    }
```
Comment (hadleynet) [inline src/main/java/org/mitre/synthea/world/geography/Demographics.java:529]: You could replace this for loop with: map.replaceAll((key, value) -> value == null || value.isNaN() ? 0.0 : value)

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java [hunk 2]
```diff
@@ -502,12 +506,43 @@ private static Demographics csvLineToDemographics(Map<String,String> line) {
...
+      if (value != 0)  {
+        allZero = false;
+        break;
+      }
+    }
+    if (allZero) {
+      Double value = 1.0 / Double.valueOf(map.size());
+      for (String key : map.keySet()) {
+        map.put(key, value);
+      }
```
Comment (hadleynet) [inline src/main/java/org/mitre/synthea/world/geography/Demographics.java:542]: You could replace this for loop with map.replaceAll((key, oldValue) -> value)

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java [hunk 3]
```diff
@@ -502,12 +506,43 @@ private static Demographics csvLineToDemographics(Map<String,String> line) {
...
+    // Now check if all values are zero
+    boolean allZero = true;
+    for (Double value : map.values()) {
+      if (value != 0)  {
+        allZero = false;
+        break;
+      }
+    }
+    if (allZero) {
+      Double value = 1.0 / Double.valueOf(map.size());
```
Comment (hadleynet) [inline src/main/java/org/mitre/synthea/world/geography/Demographics.java:539]: I don't think you need the Double.valueOf here

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java [hunk 4]
```diff
@@ -502,12 +506,43 @@ private static Demographics csvLineToDemographics(Map<String,String> line) {
...
+   * If the values are all zero, an equally weighted uniform distribution
+   * will be set.
+   * @param map The map to check.
+   */
+  private static void nonZeroDefaults(Map<String, Double> map) {
+    // Any null or nan values should be zero
+    for (String key : map.keySet()) {
+      Double value = map.get(key);
+      if (value == null || value.isNaN()) {
+        map.put(key, Double.valueOf(0));
```
Comment (hadleynet) [inline src/main/java/org/mitre/synthea/world/geography/Demographics.java:527]: Why not map.put(key, 0.0)?

**Review by hadleynet — CHANGES_REQUESTED** [review]

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java (hunk 1 above)
Comment (jawalonoski) [reply src/main/java/org/mitre/synthea/world/geography/Demographics.java]: Done.

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java (hunk 2 above)
Comment (jawalonoski) [reply src/main/java/org/mitre/synthea/world/geography/Demographics.java]: Done.

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java (hunk 3 above)
Comment (jawalonoski) [reply src/main/java/org/mitre/synthea/world/geography/Demographics.java]: Done.

File: src/main/java/org/mitre/synthea/world/geography/Demographics.java (hunk 4 above)
Comment (jawalonoski) [reply src/main/java/org/mitre/synthea/world/geography/Demographics.java]: Done.

**Review by hadleynet — APPROVED** [review]

**Pull request merged**